src/openrct2/ride/vehicle.c

```diff
@@ -1852,7 +1852,14 @@ static int ride_get_train_index_from_vehicle(rct_ride* ride, uint16 spriteIndex)
     while (ride->vehicles[trainIndex] != spriteIndex)
     {
         trainIndex++;
-        if (trainIndex >= ride->num_vehicles || trainIndex >= countof(ride->vehicles))
+        if (trainIndex >= ride->num_vehicles)
```

Comment:
It's not. They just force a couple trains from a different track onto this one so it will vary with each park